gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/KafkaReportingFormats.java

```diff
@@ -17,87 +17,170 @@
+    public void buildMetricsScheduledReporter(String brokers, String topic, Properties properties) throws IOException {
+      KafkaReporter.Builder builder = KafkaReporter.BuilderFactory.newBuilder();
+      builder.build(brokers, topic, properties);
+    }
+
+    @Override
+    public ScheduledReporter buildEventsScheduledReporter(String brokers, String topic, MetricContext context, Properties properties) throws IOException {
+       KafkaEventReporter.Builder builder = KafkaEventReporter.Factory.forContext(context);
+       //builder.withConfig(getEventsConfig(properties));
+       String pusherClassName = properties.containsKey(PusherUtils.KAFKA_PUSHER_CLASS_NAME_KEY_FOR_EVENTS)
+          ? properties.getProperty(PusherUtils.KAFKA_PUSHER_CLASS_NAME_KEY_FOR_EVENTS)
+          : properties.getProperty(PusherUtils.KAFKA_PUSHER_CLASS_NAME_KEY,
+              PusherUtils.DEFAULT_KAFKA_PUSHER_CLASS_NAME);
+       builder.withPusherClassName(pusherClassName);
+
+      Config allConfig = ConfigUtils.propertiesToConfig(properties);
+      // the kafka configuration is composed of the metrics reporting specific keys with a fallback to the shared
+      // kafka config
+      Config kafkaConfig = ConfigUtils.getConfigOrEmpty(allConfig,
+          PusherUtils.METRICS_REPORTING_KAFKA_CONFIG_PREFIX).withFallback(ConfigUtils.getConfigOrEmpty(allConfig,
+          ConfigurationKeys.SHARED_KAFKA_CONFIG_PREFIX));
+
+      builder.withConfig(kafkaConfig);
+
+       return builder.build(brokers, topic);
+    }
+  },
+  PLAIN_OBJECT() {
+
+    @Override
+    public void buildMetricsScheduledReporter(String brokers, String topic, Properties properties) throws IOException {
+
+      KeyValueMetricObjectReporter.Builder<?> builder = KeyValueMetricObjectReporter.Factory.newBuilder();
+      builder.namespaceOverride(KafkaAvroReporterUtil.extractOverrideNamespace(properties));
+      Config allConfig = ConfigUtils.propertiesToConfig(properties);
+      Config config = ConfigUtils.getConfigOrEmpty(allConfig, ConfigurationKeys.METRICS_REPORTING_CONFIGURATIONS_PREFIX).withFallback(allConfig);
+      builder.build(brokers, topic, config);
+
+    }
+
+    @Override
+    public ScheduledReporter buildEventsScheduledReporter(String brokers, String topic, MetricContext context, Properties properties) throws IOException {
+
+      KeyValueEventObjectReporter.Builder<?> builder = KeyValueEventObjectReporter.Factory.forContext(context);
+      Config allConfig = ConfigUtils.propertiesToConfig(properties);
+      Config config = ConfigUtils.getConfigOrEmpty(allConfig, ConfigurationKeys.METRICS_REPORTING_EVENTS_CONFIGURATIONS_PREFIX).withFallback(allConfig);
+      builder.withConfig(config);
+      builder.namespaceOverride(KafkaAvroReporterUtil.extractOverrideNamespace(properties));
+      return builder.build(brokers, topic);
+    }
+  };
+
+  public abstract void buildMetricsScheduledReporter(String brokers, String topic, Properties properties) throws IOException;
```

Comment:
can use a more concise name `buildMetricsReporter`, the same applied to `buildEventsReporter`